crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,304 @@
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        match item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                let capture_method = if let Some(capture_method) = item.request.capture_method {
+                    capture_method.to_string()
+                } else {
+                    "automatic".to_string()
+                };
+                let auth_mode = if capture_method.to_lowercase() == "manual" {
+                    BluesnapPaymentStatus::AuthOnly
+                } else {
+                    BluesnapPaymentStatus::AuthCapture
+                };
+                let payment_request = Self {
+                    amount: item.request.amount.to_string(),
+                    credit_card: Card {
+                        card_number: ccard.card_number.clone(),
+                        expiration_month: ccard.card_exp_month.peek().clone(),
+                        expiration_year: ccard.card_exp_year.peek().clone(),
+                        security_code: ccard.card_cvc.peek().clone(),
+                    },
+                    currency: item.request.currency.to_string(),
+                    soft_descriptor: item.description.clone(),
+                    card_transaction_type: auth_mode,
+                    card_holder_info: CardHolderInfo {
+                        first_name: ccard.card_holder_name.peek().clone(),
+                    },
+                };
+                Ok(payment_request)
+            }
+            _ => Err(
+                errors::ConnectorError::NotImplemented("Current Payment Method".to_string()).into(),
+            ),
+        }
+    }
+}
+
+// Auth Struct
+pub struct BluesnapAuthType {
+    pub(super) api_key: String
+}
+
+impl TryFrom<&types::ConnectorAuthType> for BluesnapAuthType  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+    }
+}
+// PaymentsResponse
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BluesnapPaymentStatus {
+    #[default]
+    AuthOnly,
+    AuthCapture,
+    AuthReversal,
+    Capture,
+}
+
+impl From<BluesnapPaymentStatus> for enums::AttemptStatus {
+    fn from(item: BluesnapPaymentStatus) -> Self {
+        match item {
+            BluesnapPaymentStatus::AuthOnly => Self::Authorized,
+            BluesnapPaymentStatus::AuthCapture => Self::Charged,
+            BluesnapPaymentStatus::AuthReversal => Self::Voided,
+            BluesnapPaymentStatus::Capture => Self::Charged,
+        }
+    }
+}
```

Comment:
Refer [authorizedotnet](https://github.com/juspay/hyperswitch/tree/authorize-net-features/crates/router/src/connector/authorizedotnet)
In `transformers.rs` line 376